plugins/anomalydetection/src/plugin.cpp

```diff
@@ -0,0 +1,582 @@
+        m_tty = m_thread_table.get_field(t.fields(), "tty", st::SS_PLUGIN_ST_UINT32);
+        m_cwd = m_thread_table.get_field(t.fields(), "cwd", st::SS_PLUGIN_ST_STRING);
+    }
+    catch(falcosecurity::plugin_exception e)
+    {
+        m_lasterr = "cannot init libs' thread table info fields: '" + std::string(e.what());
+        return false;
+    }
+
+    //////////////////////////
+    // Init sketches
+    //////////////////////////
+
+    // Init the plugin managed state table holding the count min sketch estimates for each behavior profile
+    if (m_rows_cols.size() == m_n_sketches)
+    {
+        for (uint32_t i = 0; i < m_n_sketches; ++i)
+        {
+            uint64_t rows = m_rows_cols[i][0];
+            uint64_t cols = m_rows_cols[i][1];
+            m_count_min_sketches.push_back(std::make_unique<plugin::anomalydetection::num::cms<uint64_t>>(rows, cols));
+        }
+    } else if (m_gamma_eps.size() == m_n_sketches && m_rows_cols.empty())
+    {
+        for (uint32_t i = 0; i < m_n_sketches; ++i)
+        {
+            double gamma = m_gamma_eps[i][0];
+            double eps = m_gamma_eps[i][1];
+            m_count_min_sketches.push_back(std::make_unique<plugin::anomalydetection::num::cms<uint64_t>>(gamma, eps));
+        }
+    } else
+    {
+        return false;
+    }
+
+    return true;
+}
+
+//////////////////////////
+// Extract capability
+//////////////////////////
+
+std::vector<falcosecurity::field_info> anomalydetection::get_fields()
+{
+    using ft = falcosecurity::field_value_type;
+    const falcosecurity::field_info fields[] = {
+            {ft::FTYPE_UINT64, "anomaly.count_min_sketch", "Count Min Sketch Estimate",
+             "Count Min Sketch Estimate according to the specified behavior profile for a predefined set of {syscalls} events. Access different behavior profiles/sketches using indices. For instance, anomaly.count_min_sketch[0] retrieves the first behavior profile defined in the plugins' `init_config`."},
+            {ft::FTYPE_STRING, "anomaly.count_min_sketch.profile", "Behavior Profile Concatenated String",
+             "Concatenated string according to the specified behavior profile (not preserving original order). Access different behavior profiles using indices. For instance, anomaly.count_min_sketch.profile[0] retrieves the first behavior profile defined in the plugins' `init_config`."},
+    };
+    const int fields_size = sizeof(fields) / sizeof(fields[0]);
+    static_assert(fields_size == ANOMALYDETECTION_FIELD_MAX, "Wrong number of anomaly fields.");
+    return std::vector<falcosecurity::field_info>(fields, fields + fields_size);
+}
+
+bool anomalydetection::extract(const falcosecurity::extract_fields_input& in)
+{
+    auto& req = in.get_extract_request();
+    uint64_t count_min_sketch_estimate = 0;
+    // todo fix settings to allow arg index in field
+    auto index = req.get_arg_index();
```

Comment:
@jasondellaluce may I ask for some guidance re how to make the arg indices available? The idea is to support notations like `anomaly.count_min_sketch[2]` etc. It should be possible, just not sure how. Thanks in advance!

Reply:
In `std::vector<falcosecurity::field_info> anomalydetection::get_fields()` you are defining the list of available fields. In each, when doing `{...}` you're invoking their constructor and defining them one by one. The 5th (and optional) argument of that constructor is the field arg, which is a substruct you can define to represent what the argument for that field looks like: https://github.com/falcosecurity/plugin-sdk-cpp/blob/efa5564221cea590eaf727a7a26e655515848c2b/include/falcosecurity/types.h#L149

Example:
```
...
{
    ft::FTYPE_UINT64,
    "anomaly.count_min_sketch",
    "Count Min Sketch Estimate",
    "Count Min Sketch Estimate according to the specified behavior profile for a predefined set of {syscalls} events. Access different behavior profiles/sketches using indices. For instance, anomaly.count_min_sketch[0] retrieves the first behavior profile defined in the plugins' `init_config`.",
    { // field arg
        false, // key
        true,  // index
        false, // required (don't know, maybe it is -- you know better)
    }
},
...
```

After this declaration, the framework (libs) will handle all the syntax parsing for you and will provide you the user-defined argument, which you can access through `req.get_arg_index()`

Reply:
Thank you! I see it's a sub struct :upside_down_face: works like a charm now!